src/EditorFeatures/Core/Extensibility/NavigationBar/INavigationBarController.cs

```diff
@@ -7,5 +7,7 @@ namespace Microsoft.CodeAnalysis.Editor
     internal interface INavigationBarController
     {
         void Disconnect();
+
+        void SetWorkspace(Workspace newWorkspace);
```

Comment:
Did you consider alternatively just accepting the Workspace when you create the INavigationBarController and then destroying and re-creating a new NavigationBarController rather than having to make the Workspace changeable?

Reply:
(not saying this is *better* but just thinking out loud...)

Reply:
I thought about it for a bit - I think it's fine to just update the workspace as the controller seems to conceptually be for a particular buffer we know how to handle, not necessarily tied to a workspace.  And it's fine for the workspace to change for a buffer we know about (e.g. file goes in or out of misc).  Let me know if you disagree strongly though.